core/gather/driver/target-manager.js

```diff
@@ -185,8 +235,12 @@ class TargetManager extends ProtocolEventEmitter {
     this._targetIdToTargets = new Map();
 
     this._rootCdpSession.on('Page.frameNavigated', this._onFrameNavigated);
+    this._rootCdpSession.on('Runtime.executionContextCreated', this._onExecutionContextCreated);
+    this._rootCdpSession.on('Runtime.executionContextDestroyed', this._onExecutionContextDestroyed);
+    this._rootCdpSession.on('Runtime.executionContextsCleared', this._onExecutionContextsCleared);
 
     await this._rootCdpSession.send('Page.enable');
+    await this._rootCdpSession.send('Runtime.enable');
```

Comment:
Will this flush `Runtime.executionContextCreated` events for execution contexts on the page when the run started?

Reply:
looking around the [code](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/inspector/v8-inspector-impl.cc;bpv=1;bpt=1?q=executionContextCreated%20f:cc%20-f:out&ss=chromium) ... i don't think so.

Reply:
DevTools populates the execution context dropdown though, so it must be possible somehow

Reply:
That might be a problem for timespan/snapshot mode. Is there some other way to see contexts created before the run?